stl/parser.py

```diff
@@ -681,6 +697,7 @@ def p_empty(p):
     p[0] = None
 
   def p_error(p):
+    parse_env['error'] = True
```

Comment:
We do, because the parser will fail silently if any exceptions are thrown during a production; p_error won't be called in this case: http://www.dabeaz.com/ply/ply.html#ply_nn35

So we need some way of propagating errors back to the original caller. A return argument seemed like the most reasonable approach. Also in a few other CLs I'm using parse_env to store some context between production rules; the plan is to replace the `module_dict` argument, which allows us to support defining a module across several files.